tiled/authenticators.py

```diff
@@ -98,20 +93,18 @@ def __init__(self, service="login", confirmation_message=""):
         self.confirmation_message = confirmation_message
         # TODO Try to open a PAM session.
 
-    async def authenticate(self, username: str, password: str) -> UserSessionState:
+    async def authenticate(self, username: str, password: str) -> UserSessionState | None:
         import pamela
 
         try:
             pamela.authenticate(username, password, service=self.service)
+            return UserSessionState(username, {})
```

Comment:
IMO this is clearer than try/except/else: if pamela.authenticate doesn't throw an exception, we go from line 100 to line 101, rather than to line 110, and the happy path all lives together